music/src/ddsp/constants.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ *
+ * @license
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const MIN_VRAM = 50; // used for memory check
+
+export enum MODEL {
+  VIOLIN = 'violin',
+  TENOR_SAXOPHONE = 'tenor_saxophone',
+  TRUMPET = 'trumpet',
+  FLUTE = 'flute',
+}
+
+export const MODEL_URL =
+  'https://storage.googleapis.com/magentadata/js/checkpoints/ddsp';
+
+export const LOWEST_LD = -120;
+export const CROSSFADE_DURATION = 1;
+export const OUTPUT_SAMPLE_RATE = 48000;
+
+export const PRESET_MODELS = [
```

Comment:
More about the leaky abstraction / constructor. Taking a new look at this, I don't think having these as constants in the javascript code makes a lot of sense.

We'd like the ability to let people train their own models soon, and this would require them to add these values to the magenta-js code which doesn't work well for dropping in new models at runtime. 

I think it might make sense to have these in a json object (stored on cloud) that is read as well as the model weights. That way we can have our training scripts create these extra json files as well, and then users can just point to the URL of the directory with all the model info in it. 

I think ideally, the model constructor would take the URL and these values in it's constructor, but then override these values with values from the JSON if it's found.

We could do this as a later PR if you want, but I think it's important to do, and would keep the Model class from being leaky.

Reply:
i see what you mean now with the constructor, perhaps we can have it in both places. these options will be read in the constructor and can be overwritten in the synthesize method.

i agree with taking it out of the constants file and moving them somewhere else.   

now that i take a look at this again, i realize that there are no default values if the checkpointUrl doesn't match any of these preset models. perhaps each one of these values need to be required parameters in the constructor and we can list the preset models' values in documentation.

Reply:
yah, I think it makes sense to override with values read them from jsons (that we can create for new datasets and new models as well).

i.e. Order of overwriting:
====================
--> Constructor Defaults
--> Constructor Args
---> Values loaded from json read in constructor

Reply:
I've moved these values out of the constants and created a new README to provide these values since we don't have a way to load them from a json alongside the model for now.